### ISSUE TITLE
Bringin' back the shouting rainbows

### ISSUE THREAD

**Issue by jenesaisdiq** [particle-iot/particle-cli#92]
As a user of the CLI,
I would like to experience rainbow-colored joy upon successful setup of a device,
because it used to do that, and also I like pretty lights.

**Comment by monkbroc** [comment]
![](http://media.giphy.com/media/WC0DCFkLq4faU/giphy.gif)

**Comment by jenesaisdiq** [comment]
^ this guy knows what I'm talking about

**Comment by towynlin** [comment]
Adding my support :+1: to help disambiguate which Photon I'm talking to when there are multiple in the room and I'm unsure which device has which SSID.

**Comment by kennethlimcp** [comment]
Ok i tested this briefly and `particle nyan deviceID on/off` works fine. `all` does not work with the use of interactive prompt for `listDevices`. Should not be too tough a fix. 

@jenesaisdiq can you try that? `deviceID` or `deviceName` worked for me

**Comment by brycekahle** [comment]
@kennethlimcp Can you file that as a separate issue since this is about the setup process?

**Comment by tacoe** [comment]
`particle nyan` should probably print some feedback. I tried it, nothing happened, then failed to mentally connect the dots when I later found my Photon going crazy with rainbows :smile:

**Comment by emilyrose** [comment]
@tacoe haha, good point!

**Comment by jenesaisdiq** [comment]
bump

**Comment by emilyrose** [comment]
:+1:

**Comment by m-mcgowan** [comment]
I feel I need more details before being able to implement this:

- when do the rainbows start?
- when do they stop?

The initial description implies that the device shouts rainbows after the device is setup...but that seems like it's too late?

When devices are first powered up, they enter listening mode. After a device has been setup, the device is no longer in listening mode, and is ideally breathing cyan - that could serve as an indicator that the device was setup? Also since it's no longer in listening mode the device is no longer available for setup via WiFi.

**Comment by jenesaisdiq** [comment]
I defer to @dmiddlecamp on what it used to do, but I'd say it does nyan when it first successfully connects to the cloud, in place of breathing cyan. Did the original stop after a while David? I feel like 10 seconds of nyan's probably good...

**Comment by technobly** [comment]
Is this something the CLI used to do during `particle setup` ?  Otherwise it's kind of a complicated thing that Cloud would have to work out the logic for... i.e.,  going from no valid key to valid key, send the nyan event on, then turn it off after 10 seconds.  If the CLI does this it would be less `state`/`timer` info that the Cloud has to manage, and could run every time `particle setup` was completed successfully.

Regarding @towynlin 's comment:
>Adding my support 👍 to help disambiguate which Photon I'm talking to when there are multiple in the room and I'm unsure which device has which SSID.

I think this is handled by the new Signal / Stop Signaling in the Web IDE, but not necessarily something that is accomplished by running `particle setup` on a room full of devices... unless the idea was that you start the rainbows DURING setup to help identify which blinking blue Photon you selected from a list of available photons during `particle setup`.  But if that was the idea then the logic changes a bit.

Use Case 1: Rainbow Joy after setup completes, YAY 🌈  ☁️ 
Use Case 2: Rainbow Joy at the beginning of setup to help identify which blinking blue photon we are setting up.  At a hackathon having both of these happening though would make things fairly confusing.

A 3rd User Case already exists as well: identify a Photon with the Signal button in the Web IDE, but only after it's setup and connected to the Cloud.

:thinking:

**Comment by jenesaisdiq** [comment]
I'm guessing that it was triggered by the CLI and by the mobile apps back in the day, upon successful setup and connection to the cloud. I do like the observation that for ID'ing which one you're working with, it's much wiser to signal at the beginning of the process rather than at the end. In a room full of devices it's chaos anyway and neither helps... for that scenario we talked about having a handful of different LED codes and randomly distributing them at manufacture time, or having the CLI/app generate one and show you which it was.

**Comment by dmiddlecamp** [comment]
If I remember correctly, the shouting rainbows happened after you claimed / named the device, and was accompanied by happy ascii art.  Particle setup won't setup a room full of devices anymore, so that seems like less of a concern.  It's more to congratulate the user on finishing setup, and help build excitement about using the device.

It's also very useful for identifying a particular device later on, but that's less important during setup  / that's a different problem.

**Comment by monkbroc** [comment]
BTW it still works for Cores. I set up a Core this weekend and the CLI said it would start shouting rainbows and it did :rainbow:
